File: src/main/zhangliang/countprimes/CountPrimes.java
package main.zhangliang.countprimes;

public class CountPrimes {

	public static void main(String[] args) {
		System.out.println(new CountPrimes().countPrimes(10));
	}

	public int countPrimes(int n) {
		if (n <= 2)
			return 0;
		int count = 1;
		int m = -1;
		boolean isprime = false;
		for (int i = 3; i < n; i += 2) {
			m = i;
			isprime = true;
			for (int j = 3; j < m; j += 2) {
				if (i % j == 0) {
					isprime = false;
					break;
				} else {
					m = i / j ;
				}
			}
			if (isprime) {
				count++;
			}
		}
		return count;
	}
}

File: src/main/houpo/countPrimes.js
/**
 * Your runtime beats 63.91 % of javascript submissions.
 * https://leetcode.com/problems/count-primes/description/
 * @param {number} n
 * @return {number}
 */
var countPrimes = function(n) {
    var arr = [];
    var sum = 0 ;
    for(var i=2;i<n;i++){
        if(isPrime(arr, i)){
            sum ++;
        }
    }
    return sum;
};
var isPrime = function(arr, i){
    if(i == 2){
        arr.push(i);
        return true;
    }
    var half = Math.sqrt(i);
    for(var j=0;j<arr.length;j++){
        if(i%arr[j] == 0){
            return false;
        }
        if(arr[j]>=half){
            break;
        }
    }
    arr.push(i);
    return true;
}
